pages/add/index.js

```diff
@@ -0,0 +1,37 @@
+import Form from "../../components/Form";
+import { nanoid } from "nanoid";
+export default function NewFacilities({ setFacilities }) {
+  function addFacilities(
+    name,
+    category,
+    adress,
+    link,
+    target,
+    requirements,
+    filterCriteria,
+    tel
+  ) {
+    setFacilities((oldFacilities) => [
+      ...oldFacilities,
+      {
+        id: nanoid(),
+        name,
+        category,
+        adress,
+        link,
+        target,
+        requirements,
+        image: "/icons/app.png",
+        filterCriteria,
+        tel,
+        isBookmarked: false,
+      },
+    ]);
+  }
```

Comment:
I suggest to move this function to the location of your `facilities` state (in your `_app.js`?). You then need to pass down this function as a prop instead of `setFacilities`. In this way, you will have most functions in one place and it is easier to extract them later on, when you want to refactor your code.

As in general it is better to update the state directly, where it is, instead of passing down the "set-function" as a prop.